gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -0,0 +1,501 @@
+                address: address.to_string(),
+            })
+            .map_err(|e| LightningRpcError::FailedToGetFundingAddress {
+                failure_reason: e.to_string(),
+            })
+    }
+
+    async fn open_channel(
+        &self,
+        pubkey: secp256k1::PublicKey,
+        host: String,
+        channel_size_sats: u64,
+        push_amount_sats: u64,
+    ) -> Result<EmptyResponse, LightningRpcError> {
+        let push_amount_msats_or = if push_amount_sats == 0 {
+            None
+        } else {
+            Some(push_amount_sats * 1000)
+        };
+
+        self.node
+            .connect_open_channel(
+                pubkey,
+                SocketAddress::from_str(&host).map_err(|e| {
+                    LightningRpcError::FailedToConnectToPeer {
+                        failure_reason: e.to_string(),
+                    }
+                })?,
+                channel_size_sats,
+                push_amount_msats_or,
+                None,
+                true,
+            )
+            .map_err(|e| LightningRpcError::FailedToOpenChannel {
+                failure_reason: e.to_string(),
+            })?;
+
+        Ok(EmptyResponse {})
+    }
+
+    async fn close_channels_with_peer(
+        &self,
+        pubkey: secp256k1::PublicKey,
+    ) -> Result<CloseChannelsWithPeerResponse, LightningRpcError> {
+        let mut num_channels_closed = 0;
+
+        for channel_with_peer in self
+            .node
+            .list_channels()
+            .iter()
+            .filter(|channel| channel.counterparty_node_id == pubkey)
+        {
+            if self
+                .node
+                .close_channel(&channel_with_peer.user_channel_id, pubkey)
+                .is_ok()
+            {
+                num_channels_closed += 1;
+            }
+        }
+
+        Ok(CloseChannelsWithPeerResponse {
+            num_channels_closed,
+        })
+    }
+
+    async fn list_active_channels(&self) -> Result<Vec<ChannelInfo>, LightningRpcError> {
+        let mut channels = Vec::new();
+
+        for channel_details in self
+            .node
+            .list_channels()
+            .iter()
+            .filter(|channel| channel.is_channel_ready)
+        {
+            channels.push(ChannelInfo {
+                remote_pubkey: channel_details.counterparty_node_id.to_string(),
+                channel_size_sats: channel_details.channel_value_sats,
+                outbound_liquidity_sats: channel_details.outbound_capacity_msat / 1000,
+                inbound_liquidity_sats: channel_details.inbound_capacity_msat / 1000,
+                short_channel_id: match channel_details.funding_txo {
+                    Some(funding_txo) => self.outpoint_to_scid(funding_txo).await.unwrap_or(0),
+                    None => 0,
+                },
+            });
+        }
+
+        Ok(channels)
+    }
+}
+
+fn create_short_channel_id(block_height: u32, transaction_index: u32, output_index: u32) -> u64 {
```

Comment:
We have a very similar function in the CLN extension called `scid_to_u64`. Might be worth to extract somewhere common

Reply:
I learned that the `lightning` crate exposes an [`scid_from_parts`](https://docs.rs/lightning/0.0.123/lightning/util/scid_utils/fn.scid_from_parts.html) function that does exactly what this function here does, so I removed this one. As for `scid_to_u64`, cln_rpc's `ShortChannelId` struct is represented using a `u64` under the hood, so I made this PR to expose that so we won't need to do any conversion there:

https://github.com/ElementsProject/lightning/pull/7450

Reply:
Perfect